.github/actions/repo-metadata/action.yaml

```diff
@@ -0,0 +1,39 @@
+name: repo-metadata
+description: Retrieve repository metadata using the GitHub API
+inputs:
+  gh-token:
+    description: GitHub token to query the API
+    required: true
+outputs:
+  fork:
+    description: Repository fork status
+  private:
+    description: Repository privacy status
+  default-branch:
+    description: Default branch of the repository
```

Comment:
is this just the branch-name (e.g. `master`), or the full ref (e.g. `refs/heads/master`)?